packages/editor/src/components/document-outline/index.js

```diff
@@ -92,7 +83,8 @@ export const DocumentOutline = ( { blocks = [], title, onSelect, isTitleSupporte
 					<DocumentOutlineItem
 						level={ __( 'Title' ) }
 						isValid
-						onClick={ focusTitle }
+						close={ close }
+						target={ '#' + titleNode.id }
```

Comment:
Ah, I see why the selector is used now, though this still seems worrying 🤔

Side note: we can use template strings here to clean it up a bit 🤷‍♂️:

```suggestion
						target={ `#${ titleNode.id }` }
```